src/api/client.ts

```diff
@@ -0,0 +1,139 @@
+          const { status, data } = error.response;
+          
+          logger.debug(`Received error response: ${status} - ${safeStringify(data)}`);
+          
+          if (status === 401) {
+            logger.error('Authentication failed. Please check your API key.');
+          } else if (status === 403) {
+            logger.error('You do not have permission to perform this action.');
+          } else if (status === 404) {
+            logger.error('Resource not found.');
+          } else {
+            logger.error(`API Error (${status}): ${data.message || safeStringify(data)}`);
+          }
+        } else if (error.request) {
+          logger.error('No response received from the server. Please check your internet connection.');
+          logger.debug(`Request details: ${safeStringify(error.request).substring(0, 200)}...`);
+        } else {
+          logger.error(`Error: ${error.message}`);
+        }
+        
+        return Promise.reject(error);
+      }
+    );
+  }
+
+  async get<T>(url: string, config?: AxiosRequestConfig): Promise<T> {
+    try {
+      logger.debug(`GET request to: ${url}`);
+      const response = await this.client.get<T>(url, config);
+      return response.data;
+    } catch (error) {
+      logger.debug(`GET request failed: ${error instanceof Error ? error.message : String(error)}`);
+      throw error;
+    }
+  }
+
+  async post<T>(url: string, data?: any, config?: AxiosRequestConfig): Promise<T> {
+    try {
+      logger.debug(`POST request to: ${url}`);
+      const response = await this.client.post<T>(url, data, config);
+      return response.data;
+    } catch (error) {
+      logger.debug(`POST request failed: ${error instanceof Error ? error.message : String(error)}`);
+      throw error;
+    }
+  }
+
+  async put<T>(url: string, data?: any, config?: AxiosRequestConfig): Promise<T> {
+    try {
+      logger.debug(`PUT request to: ${url}`);
+      const response = await this.client.put<T>(url, data, config);
+      return response.data;
+    } catch (error) {
+      logger.debug(`PUT request failed: ${error instanceof Error ? error.message : String(error)}`);
+      throw error;
+    }
+  }
+
+  async delete<T>(url: string, config?: AxiosRequestConfig): Promise<T> {
+    try {
+      logger.debug(`DELETE request to: ${url}`);
+      const response = await this.client.delete<T>(url, config);
+      return response.data;
+    } catch (error) {
+      logger.debug(`DELETE request failed: ${error instanceof Error ? error.message : String(error)}`);
+      throw error;
+    }
+  }
+
+  async patch<T>(url: string, data?: any, config?: AxiosRequestConfig): Promise<T> {
+    try {
+      logger.debug(`PATCH request to: ${url}`);
+      const response = await this.client.patch<T>(url, data, config);
+      return response.data;
+    } catch (error) {
+      logger.debug(`PATCH request failed: ${error instanceof Error ? error.message : String(error)}`);
+      throw error;
+    }
+  }
+}
```

Comment:
Use `ofetch.create`: https://github.com/unjs/ofetch?tab=readme-ov-file#%EF%B8%8F-create-fetch-with-default-options

Reply:
So I should not use axios and use `ofetch` when creating the apiClient? Does this avoid the need to generate a new function for each API call? I might be missing the context on why or is it that we should opt of fetch over axios?

Reply:
I can switch this in next release.

Reply:
Yes, we can avoid a lot of boilerplate code with typing. It's not urgent, but we can think of it as a good thing that makes life easier.